emcc.py

```diff
@@ -1077,43 +1075,29 @@ def add_link_flag(i, f):
         skip = True
 
       if not arg.startswith('-'):
+        # we already removed -o <target>, so all these should be inputs
+        newargs[i] = ''
         # os.devnul should always be reported as existing but there is bug in windows
         # python before 3.8:
         # https://bugs.python.org/issue1311
         if not os.path.exists(arg) and arg != os.devnull:
           exit_with_error('%s: No such file or directory ("%s" was expected to be an input file, based on the commandline arguments provided)', arg, arg)
         file_suffix = get_file_suffix(arg)
-        if file_suffix in SOURCE_ENDINGS + DYNAMICLIB_ENDINGS + ASSEMBLY_ENDINGS + HEADER_ENDINGS or building.is_ar(arg):
-          # we already removed -o <target>, so all these should be inputs
-          newargs[i] = ''
-          if file_suffix in SOURCE_ENDINGS or (has_dash_c and file_suffix == '.bc'):
-            input_files.append((i, arg))
-          elif file_suffix in HEADER_ENDINGS:
-            input_files.append((i, arg))
-            has_header_inputs = True
-          elif file_suffix in ASSEMBLY_ENDINGS or building.is_bitcode(arg) or building.is_ar(arg):
-            input_files.append((i, arg))
-          elif building.is_wasm(arg):
-            input_files.append((i, arg))
-          elif file_suffix in (STATICLIB_ENDINGS + DYNAMICLIB_ENDINGS):
-            # if it's not, and it's a library, just add it to libs to find later
-            libname = unsuffixed_basename(arg)
-            for prefix in LIB_PREFIXES:
-              if not prefix:
-                continue
-              if libname.startswith(prefix):
-                libname = libname[len(prefix):]
-                break
-            libs.append((i, libname))
-        elif file_suffix in STATICLIB_ENDINGS:
-          if not building.is_ar(arg):
-            if building.is_bitcode(arg):
-              message = arg + ': File has a suffix of a static library ' + str(STATICLIB_ENDINGS) + ', but instead is an LLVM bitcode file! When linking LLVM bitcode files use .bc or .o.'
-            else:
-              message = arg + ': Unknown format, not a static library!'
-            exit_with_error(message)
+        if file_suffix in HEADER_ENDINGS:
+          has_header_inputs = True
```

Comment:
the old code used to add header inputs to `input_files`, but  this no longer does I think - is that intentional?

Reply:
oops, that was in my original patch.  restored.